packages/flutter/lib/src/foundation/tree_diagnostics_mixin.dart

```diff
@@ -2,9 +2,1438 @@
+  /// null. If `quoted` is true, the `value` is enclosed
+  /// in double quotes.
+  factory DiagnosticsNode.stringProperty(String name, String value, {
+    bool showNull: true,
+    String header,
+    bool showName: true,
+    bool hidden: false,
+    bool quoted: false,
+  }) {
+    header ??= value;
+    if (quoted)
+      header = '"$header"';
+
+    return new _LeafDiagnosticsValue(
+      name: name,
+      header: header,
+      object: value,
+      showNull: showNull,
+      showName: showName,
+      hidden: hidden,
+    );
+  }
+
+  /// Property with a [bool] value.
+  ///
+  /// If `showNull` is true, the property is shown even if its `value` is null.
+  /// `header` overrides the normal description of the property value.
+  /// `showNull` indicates whether the property should still be shown if its
+  /// `value` is null. `showName indicates whether the property name should be
+  /// shown. `hidden` specifies whether the property should be hidden or shown.
+  /// `nullDescription` provides a description of the property value if it is
+  /// null.
+  factory DiagnosticsNode.boolProperty(String name, bool value, {
+    String header,
+    bool showNull: true,
+    bool showName: true,
+    bool hidden: false,
+    String nullDescription,
+  }) {
+    if (value == null)
+      header = nullDescription;
+
+    return new _LeafDiagnosticsValue(
+      name: name,
+      header: header ?? value.toString(),
+      object: value,
+      showNull: showNull,
+      hidden: hidden,
+      showName: showName,
+    );
+  }
+
+  /// A boolean flag property that is shown as `name` if value is true and is
+  /// true and hidden otherwise.
+  factory DiagnosticsNode.flagProperty(String name, bool value) {
+    return new _LeafDiagnosticsValue(
+      name: name,
+      header: value == true ? name : '',
+      object: value,
+      hidden: value != true,
+      showName: false);
+  }
+
+  /// An object property with a `comment` describing the value appended after
+  /// the description of the `value.
+  factory DiagnosticsNode.objectPropertyWithComment(String name, Object value, {
+    @required String comment,
+  }) {
+    return new _LeafDiagnosticsValue(
+      name: name,
+      object: value,
+      header: '$value ($comment)',
+    );
+  }
+
+  /// Creates a property with a [double] value.
+  ///
+  ///  If `showNull` is true, the property is shown even if its `value` is null.
+  /// `showNull` indicates whether the property should still be shown if its
+  /// `value` is null. `hidden` specifies whether the property should be hidden
+  /// or shown. `nullDescription` provides a description of the property value
+  /// if it is null.
+  /// If specified, exactly `fractionDigits` after the decimal point are shown.
+  /// If specified, `unit` describes the unit for the value (e.g. px).
+  factory DiagnosticsNode.doubleProperty(String name, double value, {
+    bool showNull: true,
+    int fractionDigits,
```

Comment:
By default double.toString() displays as many decimal digits as there are to show without showing trailing zeros. The old toString code tried to uniformly call toStringAsFixed(1) on doubles, because showing 15 decimals digits results in a mess in the output and makes it unreadable. The problem with always truncating to 1 digit is that it means you often miss bugs when the bug is "1.00000000001 isn't exactly 1.0" (since we now just show both as 1.0). We have an extreme case where this bug is very common where we actually go out of our way to detect it and report it, namely: https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/rendering/sliver.dart#L757

I would love to have this kind of magic logic more uniformly, and had been planning on factoring out all the toStringAsFixed(1) calls at some point to be able to resolve this. With this code here, you've now made that much easier. :-)

So yes, I'm fine with forcing it to 1 for now. The next step is to do something cleverer, like the code cited above.